### DISCUSSION TITLE
How to set sourceName (video file name)

### DISCUSSION THREAD

**Discussion by toufikwb** [obs-websocket-community-projects/obs-websocket-js#243] [Q&A]
first i'm a noob developer,
i want to set custom file name to the video, can i do that? if that possible please show me the code :
here is what i'm trying :
```
const obs = new OBSWebSocket();
    obs.connect({
        'sourceName' : 'newName'
    });
......
obs.send('StartRecording');
......
obs.send('StopRecording');
```
here is the link i'm trying to follow : https://github.com/Palakis/obs-websocket/blob/4.8.0/docs/generated/protocol.md#setsourcename

i'v also tried the following :
`obs.send('SetSourceName', {
      'sourceName' : 'newName'
  });
  obs.send('StartRecording');`
everything works well except that the name of the file is the datetime

**Comment by zoton2** [comment]
If you're still curious, you can set the folder the recording is stored in using [SetRecordingFolder](https://github.com/Palakis/obs-websocket/blob/4.x-current/docs/generated/protocol.md#setrecordingfolder) and set the filename using [SetFilenameFormatting](https://github.com/Palakis/obs-websocket/blob/4.x-current/docs/generated/protocol.md#setfilenameformatting). "Source name" that are you are trying to use is completely unrelated to recording file names.
